### ISSUE TITLE
Kerberos authentication succeeds when both principal and realm specified

### ISSUE THREAD

**Issue by zvonand** [ClickHouse/ClickHouse#26197]
As stated in [CH docs](https://clickhouse.tech/docs/en/operations/external-authenticators/kerberos/#enabling-kerberos-in-clickhouse), `principal and realm sections cannot be specified at the same time. The presence of both principal and realm sections will force ClickHouse to disable Kerberos authentication`. 

Despite that, one can perform authentication when both `principal` and `realm` statements are included in `config.xml`.

`preprocessed_configs/config.xml`:
```xml
<kerberos>
    <realm>EXAMPLE.COM</realm>
    <principal>HTTP/srv1@EXAMPLE.COM</principal>
</kerberos>
```

Example of a successful request that should not succeed:
```
echo 'select currentUser()' | curl --negotiate -u : 'http://clickhouse1:8123/' --data-binary @-

kerberos_user
```
(`kerberos_user` is a user who is set up to be authenticated using Kerberos)

**Comment by traceon** [comment]
The problem was introduced at some point by other PRs. Currently investigating.

**Comment by alexey-milovidov** [comment]
@traceon What does it mean "currently"?

**Comment by traceon** [comment]
@alexey-milovidov I don't remember what that meant back then, but now fixed in this: #33689
